sdk/eventhub/Microsoft.Azure.EventHubs/src/Amqp/AmqpExceptionHelper.cs

```diff
@@ -128,5 +128,18 @@ public static Exception ToMessagingContract(string condition, string message, bo
 
             return new EventHubsException(true, message);
         }
+
+        public static bool TryConvertToRetriableException(Exception exception, out EventHubsException retriableException)
+        {
+            retriableException = null;
+
+            // InvalidOperationException with 'connection is closing' message from AMQP layer is retriable.
+            if (exception is OperationCanceledException && exception.Message.IndexOf("connection is closing", StringComparison.OrdinalIgnoreCase) != - 1)
+            {
+                retriableException = new EventHubsException(true, exception);
```

Comment:
Should we consider adding some kind of message to the `EventHubsException` mentioning to check the inner for details?

Reply:
EventHubsException clones the error message of inner exception with current approach. exception.ToString() dumps the stack of inner exception as well. Any scenario you can think of that referencing inner exception can help?